### PR TITLE
Fixed support for drive letter

### PR DESCRIPTION
**Pull request by mrbean-bremen**
- fixed behavior of NormalizePath() and JoinPaths() with drive letters
- shall fix issue #83

### CODE REVIEW

File: pyfakefs/fake_filesystem.py [hunk 1]
```diff
@@ -371,6 +371,7 @@ def __init__(self, path_separator=os.path.sep):
     if path_separator != os.sep:
       self.alternative_path_separator = None
     self.is_case_sensitive = not _is_windows and sys.platform != 'darwin'
+    self.supports_drive_letter = _is_windows
```
Comment (jmcgeheeiv) [inline pyfakefs/fake_filesystem.py]: Should this be private attribute `self._supports_drive_letter`?

File: pyfakefs/fake_filesystem.py [hunk 2]
```diff
@@ -371,6 +371,7 @@ def __init__(self, path_separator=os.path.sep):
     if path_separator != os.sep:
       self.alternative_path_separator = None
     self.is_case_sensitive = not _is_windows and sys.platform != 'darwin'
```
Comment (jmcgeheeiv) [inline pyfakefs/fake_filesystem.py]: Should this be a private attribute?

File: pyfakefs/fake_filesystem.py [hunk 3]
```diff
@@ -371,6 +371,7 @@ def __init__(self, path_separator=os.path.sep):
     if path_separator != os.sep:
       self.alternative_path_separator = None
```
Comment (jmcgeheeiv) [inline pyfakefs/fake_filesystem.py]: Should this be a private attribute?

**Comment by mrbean-bremen** [comment]
I use is_case_sensitive and alternative_path_separator in tests to be able to test this under Linux (to minimize the Windows-only tests as there is no Windows test system). I can make them private and access them from the tests anyway, don't know what your policy is here, but I would prefer the tests not to be dependent on Windows.

**Pull request merged**